enzyme/Enzyme/MLIR/Implementations/ArithAutoDiffOpInterfaceImpl.cpp

```diff
@@ -50,6 +51,61 @@ struct MulFOpInterface
   }
 };
 
+struct DivFOpInterface
+    : public AutoDiffOpInterface::ExternalModel<DivFOpInterface,
+                                                arith::DivFOp> {
+  LogicalResult createForwardModeAdjoint(Operation *op, OpBuilder &builder,
+                                         MGradientUtils *gutils) const {
+    // Derivative of r = a / b -> dr = (a * db - da * b) / (b * b)
+    auto divOp = cast<arith::DivFOp>(op);
+    if (!gutils->isConstantValue(divOp)) {
+      mlir::Value res = nullptr;
+      for (int i = 0; i < 2; i++) {
+        if (!gutils->isConstantValue(divOp.getOperand(i))) {
+          Value tmp = builder.create<arith::MulFOp>(
+              divOp.getLoc(),
+              gutils->invertPointerM(divOp.getOperand(i), builder),
+              gutils->getNewFromOriginal(divOp.getOperand(1 - i)));
+          if (res == nullptr)
```

Comment:
This res null check isn't correct. For example if getOperand(0) is constant, then you'd produce da * b rather than -da * b.